npm/cmd/start_controlplane.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/Azure/azure-container-networking/common"
+	"github.com/Azure/azure-container-networking/npm"
+	npmconfig "github.com/Azure/azure-container-networking/npm/config"
+	restserver "github.com/Azure/azure-container-networking/npm/http/server"
+	"github.com/Azure/azure-container-networking/npm/metrics"
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+func newStartNPMControlplaneCmd() *cobra.Command {
+	startNPMControlplaneCmd := &cobra.Command{
+		Use:   "controlplane",
+		Short: "Starts the Azure NPM controlplane process",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			config := &npmconfig.Config{}
+			err := viper.Unmarshal(config)
+			if err != nil {
+				return fmt.Errorf("failed to load config with error: %w", err)
+			}
+
+			flags := npmconfig.Flags{
+				KubeConfigPath: viper.GetString(flagKubeConfigPath),
+			}
+
+			return startControlplane(*config, flags)
+		},
+	}
+
+	startNPMControlplaneCmd.Flags().String(flagKubeConfigPath, flagDefaults[flagKubeConfigPath], "path to kubeconfig")
+
+	return startNPMControlplaneCmd
+}
+
+func startControlplane(config npmconfig.Config, flags npmconfig.Flags) error {
+	klog.Infof("loaded config: %+v", config)
+	klog.Infof("Start NPM version: %s", version)
+
+	var err error
+
+	err = initLogging()
+	if err != nil {
+		return err
+	}
+
```

Comment:
We will need to initialize metrics here ?

Reply:
What metrics are we referring to?

Reply:
https://github.com/Azure/azure-container-networking/blob/3d1e7399f6abd72888b13c6464479f88c91cbaba/npm/cmd/start.go#L103 

this metrics initialize.

Reply:
Metrics are initialized in start.go, but initializing again couldn't hurt. I'll have to schedule a meeting to discuss plans for metrics in the fan-out model